src/hooks/prisma.ts

```diff
@@ -0,0 +1,71 @@
+import { SpansContainer, TracerGlobals } from '../globals';
+import * as logger from '../logger';
+import { getCurrentTransactionId } from '../spans/awsSpan';
+import { createPrismaSpan, extendedPrismaSpan } from '../spans/prismaSpan';
+import { getRandomId } from '../utils';
+import { safeRequire } from '../utils/requireUtils';
+
+async function queryExtension({ query, args, model, operation }) {
+  const awsRequestId: string = TracerGlobals.getHandlerInputs().context.awsRequestId;
+  const transactionId: string = getCurrentTransactionId();
+  const spanId: string = getRandomId();
+  const prismaSpan = createPrismaSpan(
+    transactionId,
+    awsRequestId,
+    spanId,
+    {
+      started: Date.now(),
+    },
+    {
+      model,
+      operation,
+      queryArgs: args,
+    }
+  );
+  SpansContainer.addSpan(prismaSpan);
+
+  try {
+    const result = await query(args);
+    const extendedSpan = extendedPrismaSpan(prismaSpan, { result, ended: Date.now() });
+    SpansContainer.addSpan(extendedSpan);
+
+    return result;
+  } catch (error) {
+    const extendedSpan = extendedPrismaSpan(prismaSpan, {
+      error,
+      ended: Date.now(),
+    });
+    SpansContainer.addSpan(extendedSpan);
+    throw error;
+  }
+}
+
+export const hookPrisma = (prismaClientLibrary: unknown | null = null) => {
+  const prismaLib = prismaClientLibrary ? prismaClientLibrary : safeRequire('@prisma/client');
+
+  if (!prismaLib) {
+    logger.debug('Prisma client not found, skipping hook');
+    return;
+  }
+
+  const OriginalConstructor = prismaLib.PrismaClient;
+
+  if (!OriginalConstructor) {
+    logger.debug('PrismaClient class not found, skipping hook');
+    return;
+  }
+
+  prismaLib.PrismaClient = function (...args) {
+    const originalInstance = new OriginalConstructor(...args);
+
+    if (typeof originalInstance.$extends !== 'function') {
+      return originalInstance;
+    }
+
+    return originalInstance.$extends({
```

Comment:
do you want to return the response from the `$extends` function or `originalInstance`?
Can you add all your code (and specifically all the *added* code) to `safeExecute`?

Reply:
> do you want to return the response from the $extends function or originalInstance?

We have to return the output of `$extends`, as it returns a new client with the extended functionality and does not mutate `originalInstance` (actually, that's why I've added [this test](https://github.com/lumigo-io/lumigo-node/blob/ce130b9fdd9525538b8a87eecac9e720ef447aa9/src/hooks/prisma.test.js#L151)).

> Can you add all your code to safeExecute?

I guess you mean something like this:
```js
return safeExecute(
  () => originalInstance.$extends({
    query: {
      $allOperations: queryExtension,
    },
  }), 
  originalInstance
);
```
to return the original instance if `$extends` failed? yeah, it definitely makes sense

Reply:
👍

Reply:
74c0301bea933dfd8e95c3e551f5a1bcb6f0046a